src/aleph/web/controllers/storage.py

```diff
@@ -230,19 +337,27 @@ async def storage_add_file(request: web.Request):
     signature_verifier = get_signature_verifier_from_request(request)
     config = get_config_from_request(request)
     grace_period = config.storage.grace_period.value
-
-    post = await request.post()
-    try:
-        file_field = post["file"]
-    except KeyError:
-        raise web.HTTPUnprocessableEntity(reason="Missing 'file' in multipart form.")
-
-    if isinstance(file_field, FileField):
-        uploaded_file: UploadedFile = MultipartUploadedFile(file_field, len(file_field.file.read()))
+    metadata = None
+    uploaded_file: Optional[UploadedFile] = None
+
+    if request.content_type == "multipart/form-data":
+        reader = await request.multipart()
+        async for part in reader:
+            if part.name == 'file':
+                uploaded_file = MultipartUploadedFile(part, MAX_FILE_SIZE)
```

Comment:
Did you forgot to move to a context manager here or forget to push a commit '-'? Maybe "context manager" is not an obvious term ^^' 

This could should look like something like this:

```python
async with MultipartUploadedFile(part, MAX_FILE_SIZE) as uploaded_file:
    await uploaded_file.read_and_validate()
```

And same line 291:

```python
async with RawUploadedFile(request=request, max_size=MAX_UNAUTHENTICATED_UPLOAD_FILE_SIZE) as uploaded_file:
    await uploaded_file.read_and_validate()
```

Reply:
Which actually wouldn't make that much sens here since uploaded_file would be deleted once you get out of the context manager.

I don't think using a context manager here make a lot of sens or the code logic needs to be changed :/

Why did you chose to use a context manager?